### PR TITLE
Remove UseServiceAccountIAM feature flag and rename feature to UseServiceAccountExternalPermissions

### PR DESCRIPTION
**Pull request by olemarkus**

### CODE REVIEW

File: pkg/featureflag/featureflag.go [hunk 1]
```diff
@@ -88,7 +88,7 @@ var (
 	// ClusterAddons activates experimental cluster-addons support
 	ClusterAddons = new("ClusterAddons", Bool(false))
 	// UseServiceAccountIAM controls whether we use pod-level IAM permissions for our system pods and kOps addons.
-	UseServiceAccountIAM = new("UseServiceAccountIAM", Bool(false))
+	UseServiceAccountIAM = new("UseServiceAccountIAM", Bool(true))
```
Comment (rifelpet) [inline pkg/featureflag/featureflag.go:91]: grep shows that this isnt used anymore so I think we can remove it as well as from `./tests/e2e/scenarios/lib/common.sh`

File: pkg/apis/kops/cluster.go [hunk 2]
```diff
@@ -304,6 +304,9 @@ type IAMSpec struct {
 	Legacy                 bool    `json:"legacy"`
 	AllowContainerRegistry bool    `json:"allowContainerRegistry,omitempty"`
 	PermissionsBoundary    *string `json:"permissionsBoundary,omitempty"`
+	// UseServiceAccountExternalPermissions determines if managed ServiceAccounts will use external permissions directly.
+	// If this is set to false, ServiceAccounts will assume external permissions from the instance it runs on.
```
Comment (johngmyers) [inline pkg/apis/kops/cluster.go:308]:
```suggestion
	// If this is set to false, ServiceAccounts will assume external permissions from the instances they run on.
```

**Review by johngmyers — CHANGES_REQUESTED** [review]

File: pkg/featureflag/featureflag.go (hunk 1 above)
Comment (johngmyers) [reply pkg/featureflag/featureflag.go:91]: Unknown feature flags are logged and otherwise ignored, so this should be removed.

Comment (olemarkus) [reply pkg/featureflag/featureflag.go]: Done.

**Review by johngmyers — APPROVED** [review]

**Pull request merged**